### PR TITLE
Improve db_fetch_array performance

### PR DESCRIPTION
**Pull request by cproensa**
Ths is #942 rebased to current master

Add a fix for a potential flaw where db_result() does not transform special values in some database types.

### CODE REVIEW

File: core/constant_inc.php [hunk 1]
```diff
@@ -693,3 +693,11 @@
...
 define( 'MANAGE_CONFIG_ACTION_CLONE', 'clone' );
 define( 'MANAGE_CONFIG_ACTION_EDIT', 'edit' );
+
+# Databse functional type identifiers.
+define( 'DB_TYPE_UNDEFINED', 0 );
+define( 'DB_TYPE_MYSQL', 1 );
+define( 'DB_TYPE_PGSQL', 2 );
+define( 'DB_TYPE_MSSQL', 3 );
+define( 'DB_TYPE_ORACLE', 4 );
+define( 'DB_TYPE_DB2', 5 );
```
Comment (atrol) [inline core/constant_inc.php]: Should be removed as we don't support DB2 any longer.

File: core/database_api.php [hunk 2]
```diff
@@ -211,63 +215,78 @@ function db_check_database_support( $p_db_type ) {
...
+			return DB_TYPE_MYSQL;
+		case 'postgres':
+		case 'postgres7':
+		case 'pgsql':
+			return DB_TYPE_PGSQL;
+		case 'mssql':
+		case 'mssqlnative':
+		case 'odbc_mssql':
+			return DB_TYPE_MSSQL;
+		case 'db2':
```
Comment (atrol) [inline core/database_api.php]: Should be removed as we don't support DB2 any longer.

File: core/database_api.php [hunk 3]
```diff
@@ -211,63 +215,78 @@ function db_check_database_support( $p_db_type ) {
...
+}
 
-	return false;
+/**
+<<<<<<< 96b648079cd45d398d59afa03490ca9d68554561
+=======
+ * Checks if the database driver is DB2
+ * @return boolean true if db2
+ */
+function db_is_db2() {
```
Comment (atrol) [inline core/database_api.php]: Should be removed as we don't support DB2 any longer.

File: core/database_api.php [hunk 4]
```diff
@@ -211,63 +215,78 @@ function db_check_database_support( $p_db_type ) {
...
+	global $g_db_functional_type;
+	return( DB_TYPE_DB2 == $g_db_functional_type );
 }
 
 /**
+>>>>>>> Implement db type constants
```
Comment (dregad) [inline core/database_api.php]: Don't forget to delete this leftover from merge/conflict too.

**Comment by cproensa** [comment]
Addressed: atrol: remove DB2, dregad: remove merge leftovers
unneeded commits have been squashed

**Pull request merged**